ibis/pandas/core.py

```diff
@@ -152,6 +138,14 @@ def execute_with_scope(expr, scope, aggcontext=None, clients=None, **kwargs):
     scope : collections.Mapping
         A dictionary mapping :class:`~ibis.expr.operations.Node` subclass
         instances to concrete data such as a pandas DataFrame.
+    localcontext : collections.Mapping
```

Comment:
What's the key and what's the value here?

Reply:
Hmm actually that could be anything that one could define as "context". Should we add more restrictions here?

Reply:
Changed it to Optional[Tuple[pd.Timestamp, pd.Timestamp]]